jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/skaffold/FilesMojoV2KotlinTest.java

```diff
@@ -0,0 +1,202 @@
+    Mockito.when(configuration1.getChild("sourceDirs")).thenReturn(sourceDirs1);
+    Mockito.when(sourceDirs1.getChildren()).thenReturn(new Xpp3Dom[] {sourceDir1});
+    Mockito.when(sourceDir1.getValue()).thenReturn(null);
+
+    Assert.assertEquals(
+        ImmutableSet.of(Paths.get("/base/src/main/kotlin")),
+        FilesMojoV2.getKotlinSourceDirectories(mavenProject));
+  }
+
+  @Test
+  public void getKotlinSourceDirectories_emptySourceDir() {
+    Mockito.when(kotlinPlugin.getExecutions()).thenReturn(Arrays.asList(pluginExecution1));
+    Mockito.when(pluginExecution1.getConfiguration()).thenReturn(configuration1);
+    Mockito.when(configuration1.getChild("sourceDirs")).thenReturn(sourceDirs1);
+    Mockito.when(sourceDirs1.getChildren()).thenReturn(new Xpp3Dom[] {sourceDir1});
+    Mockito.when(sourceDir1.getValue()).thenReturn("");
+
+    Assert.assertEquals(
+        ImmutableSet.of(Paths.get("/base/src/main/kotlin")),
+        FilesMojoV2.getKotlinSourceDirectories(mavenProject));
+  }
+
+  @Test
+  public void getKotlinSourceDirectories_relativePath() {
+    Mockito.when(kotlinPlugin.getExecutions()).thenReturn(Arrays.asList(pluginExecution1));
+    Mockito.when(pluginExecution1.getConfiguration()).thenReturn(configuration1);
+    Mockito.when(configuration1.getChild("sourceDirs")).thenReturn(sourceDirs1);
+    Mockito.when(sourceDirs1.getChildren()).thenReturn(new Xpp3Dom[] {sourceDir1});
+    Mockito.when(sourceDir1.getValue()).thenReturn("kotlin/src");
+
+    Assert.assertEquals(
+        ImmutableSet.of(Paths.get("/base/src/main/kotlin"), Paths.get("/base/kotlin/src")),
+        FilesMojoV2.getKotlinSourceDirectories(mavenProject));
+  }
+
+  @Test
+  public void getKotlinSourceDirectories_absolutePath() {
+    Mockito.when(kotlinPlugin.getExecutions()).thenReturn(Arrays.asList(pluginExecution1));
+    Mockito.when(pluginExecution1.getConfiguration()).thenReturn(configuration1);
+    Mockito.when(configuration1.getChild("sourceDirs")).thenReturn(sourceDirs1);
+    Mockito.when(sourceDirs1.getChildren()).thenReturn(new Xpp3Dom[] {sourceDir1});
+    Mockito.when(sourceDir1.getValue()).thenReturn("/absolute/src");
+
+    Assert.assertEquals(
+        ImmutableSet.of(Paths.get("/base/src/main/kotlin"), Paths.get("/absolute/src")),
+        FilesMojoV2.getKotlinSourceDirectories(mavenProject));
+  }
+
+  @Test
+  public void getKotlinSourceDirectories_complex() {
+    Mockito.when(kotlinPlugin.getExecutions())
+        .thenReturn(Arrays.asList(pluginExecution1, pluginExecution2));
+    Mockito.when(pluginExecution1.getConfiguration()).thenReturn(configuration1);
+    Mockito.when(pluginExecution2.getConfiguration()).thenReturn(configuration2);
+    Mockito.when(configuration1.getChild("sourceDirs")).thenReturn(sourceDirs1);
+    Mockito.when(configuration2.getChild("sourceDirs")).thenReturn(sourceDirs2);
+    Mockito.when(sourceDirs1.getChildren()).thenReturn(new Xpp3Dom[] {sourceDir1, sourceDir2});
```

Comment:
Not sure if techically possible, but can we use Xpp3Dom objects directly instead of mocks here?

Reply:
Indeed. Used real objects.